### DISCUSSION TITLE
How to verify input data?

### DISCUSSION THREAD

**Discussion by jet10000** [supabase/supabase#4149] [Questions] [answered]
Django has a form that can verify the data sent by the user on the server side. How does supabase do it?

**Comment by GaryAustin1** [comment, marked as answer]
Postgres has a check option for columns.  You can use simple or complex checks including regex.
https://www.postgresql.org/docs/13/ddl-constraints.html

**Reply by psirenny** [reply]
Postgres check constraints fail fast, but one usually wants validation to return a full list of errors to the user. This is suboptimal.
